### PR TITLE
Add WolframAlpha API material property queries

### PR DESCRIPTION
**Pull request by j-fletcher**
In the case of some material properties not being specified, this PR adds a function within the materials submodule to query the WolframAlpha API for a given property of a given material. The quantity is given in metric units.

This will be used by Stefano's material constructor to find properties and build OpenMC materials.

It is necessary to provide a key for access to the Wolfram Short Answers API, and free keys are limited to 2000 calls per month. Hence, CI testing can't realistically be performed at scale, but a manual testing protocol can be created.
Register for a key at developer.wolframalpha.com

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,8 @@
+certifi==2023.7.22
+charset-normalizer==3.2.0
+idna==3.4
+requests==2.31.0
+urllib3==2.0.5
```
Comment (RemDelaporteMathurin) [inline requirements.txt:5]: Looking at QueryAPI.py it seems that your only dependency here is `requests`

The others being sub-dependencies of `requests`

Please remove certifi, charset, idna, urllib3

File: materials/QueryAPI.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        App ID Keys specific to the Short Answers API may be requested via https://developer.wolframalpha.com
+        
+        Args:
+            key (str): WolframAlpha App ID
+        """
+        self.key = key
+        
+        if self.key is None:
+            ermsg = 'API key required for property lookup. Please generate a key at https://developer.wolframalpha.com'
+            raise MissingKeyError(ermsg)
```
Comment (eepeterson) [inline materials/QueryAPI.py:24]: You probably can just raise a `ValueError` here, if `key is None` that's an error in the value

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Thanks for this!

First round of comment.
I feel like we can simplify things and refactor by raising errors and avoiding default values

File: materials/QueryAPI.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        key = self.key
+
+        if compound is None:
+            result = 'Incomplete query: no compound specified'
+        elif property is None:
+            result = 'Incomplete query: no property specified'
+        else:
+            prompt = property + "%20of%20" + compound
+            url = 'http://api.wolframalpha.com/v2/result?appid=%s&i=%s&units=metric' % (
```
Comment (eepeterson) [inline materials/QueryAPI.py:46]: maybe store this url as a class constant `_URL`?

File: materials/QueryAPI.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import requests
+
+class MissingKeyError(Exception):
```
Comment (eepeterson) [inline materials/QueryAPI.py:3]: do you need a new error type?

**Review by eepeterson — CHANGES_REQUESTED** [review]
Very cool feature! some small comments here

**Comment by j-fletcher** [comment]
Comments addressed

**Review by RemDelaporteMathurin — APPROVED** [review]
🎉

**Pull request merged**